pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/LabelScript.groovy

```diff
@@ -38,10 +38,19 @@ class LabelScript extends DeclarativeAgentScript<Label> {
 
     @Override
     Closure run(Closure body) {
-        return {
+        Closure run = {
             script.node(describable?.label) {
                 CheckoutScript.doCheckout(script, describable, describable.customWorkspace, body).call()
             }
         }
+        if (describable.retries > 1) {
+            return {
+                script.retry(count: describable.retries, conditions: [script.agent(), script.nonresumable()]) {
```

Comment:
> Another possibility

https://github.com/jenkinsci/kubernetes-plugin/pull/1190/commits/c38fe0fabf27d0583c9b546aa701dc075245e72d but I guess this is enough of a corner case that we can just say you can use the explicit `retry` step.

Reply:
Not really. In the currently proposed syntax

```groovy
retries 2
```

there is no room for customization of conditions. In other words, if you as a Pipeline author wished to use different conditions, you could not use this syntax at all—you would need to explicitly include a `retry` block in `steps`.

Reply:
I think the current approach is fine for now. We can expose additional options in `RetryableDeclarativeAgent` later if other conditions are introduced or if more control is desired.

Reply:
> expose additional options […] if more control is desired

We could of course, but then you would have a disjointed experience

```groovy
retries 2
retryConditions {…something…}
```

If we _expect_ to offer customizations here then it would be better to pick a different initial syntax, like

```groovy
retry {
  count 2
}
```